smartsim/experiment.py

```diff
@@ -159,28 +161,63 @@ def __init__(
             exp_path = osp.join(getcwd(), name)
 
         self.exp_path = exp_path
+        """The path under which the experiment operate"""
 
-        self._launcher = launcher.lower()
-
-        if self._launcher == "auto":
-            self._launcher = detect_launcher()
-        if self._launcher == "cobalt":
-            raise SSUnsupportedError("Cobalt launcher is no longer supported.")
+        # TODO: Remove this! The controller is becoming obsolete
+        self._control = Controller(launcher="local")
 
-        if launcher == "dragon":
-            self._set_dragon_server_path()
+        self._active_launchers: set[LauncherProtocol[t.Any]] = set()
+        """The active launchers created, used, and reused by the experiment"""
 
-        self._control = Controller(launcher=self._launcher)
-
-        self.fs_identifiers: t.Set[str] = set()
+        self._fs_identifiers: t.Set[str] = set()
+        """Set of feature store identifiers currently in use by this
+        experiment"""
         self._telemetry_cfg = ExperimentTelemetryConfiguration()
+        """Switch to specify if telemetry data should be produced for this
+        experiment"""
+
+    def start_jobs(
+        self, job: Job, *jobs: Job, dispatcher: Dispatcher = DEFAULT_DISPATCHER
+    ) -> tuple[LaunchedJobID, ...]:
+        """Execute a collection of `Job` instances.
+
+        :param job: The job instance to start
+        :param jobs: A collection of other job instances to start
+        :param dispatcher: The dispatcher that should be used to determine how
+            to start a job based on its settings. If not specified it will
+            default to a dispatcher pre-configured by SmartSim.
+        :returns: A sequence of ids with order corresponding to the sequence of
+            jobs that can be used to query or alter the status of that
+            particular execution of the job.
+        """
 
-    def _set_dragon_server_path(self) -> None:
-        """Set path for dragon server through environment varialbes"""
-        if not "SMARTSIM_DRAGON_SERVER_PATH" in environ:
-            environ["SMARTSIM_DRAGON_SERVER_PATH_EXP"] = osp.join(
-                self.exp_path, CONFIG.dragon_default_subdir
-            )
+        def _start(job: Job) -> LaunchedJobID:
+            args = job.launch_settings.launch_args
```

Comment:
I thought we wanted the type hint because we didn't like the naming inconstancy of `launch_args` -> `builder`.

I'm still willing to add the type hint, but it would look like this:
```suggestion
            args: LaunchArguments = job.launch_settings.launch_args
```
which I don't think adds much. Willing to be talked out of this though!

Reply:
ah yeah you are right